[19.0][MIG]: l10n_es_aeat_mod303

sbiosca-s73 opened the pull request:
- Standard migration to 19.0 module l10n_es_aeat_mod303

Commit:
[IMP] l10n_es_aeat: Algo de PEP8 y refactorización de código
[MIG] l10n_es_aeat_mod303: Migración a v7 del módulo con las siguientes mejoras:
- Convenciones de la comunidad.
- PEP8 en la mayoría de código.
- Icono para el módulo.
- Adaptación de vistas a las posibilidades de la v7.
- Restringidas las cuentas bancarias a las de la compañía.

Commit:
Launchpad automatic translations update.

Commit:
[IMP] l10n_es_aeat: Mejorado mecanismo de herencia para reaprovechar más código.
[IMP] l10n_es_aeat: Añadido wizard para ver el resultado de la exportación sin necesidad de tener el módulo document.
[IMP] l10n_es_aeat: Puesta transición de estados sin workflow.
[IMP] l10n_es_aeat_mod303: Adaptado al nuevo mecanismo. Se elimina workflow anterior mediante un script de migración.

Commit:
Launchpad automatic translations update.

acysos committed:
[FIX] l10n_es_aeat: compatibilidad de la función export_to_boe para el modelo 340

Commit:
Launchpad automatic translations update.

Commit:
[FIX] l10n_es_aeat_mod130/mod303: import corregido para que valga también para la v8.

Commit:
Launchpad automatic translations update.

pedrobaeza committed:
[MIG] Movidos todos los módulos a __unported__
[MIG] Eliminados módulos no migrados a v7

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: Modelo 303 de la AEAT totalmente refactorizado

Presentación del Modelo AEAT 303
================================

* Incluye un motor genérico para obtener los totales de códigos de impuestos
  de manera parametrizada y con posibilidad de versionado por fechas
* Incluye un motor de parametrización de las exportaciones a formato BOE
  con posibilidad de versionado por fechas.
* Se incluyen las parametrizaciones de ambos motores para el 303.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Avoid errors in first initialization

oca-transbot committed:
OCA Transbot updated translations from Transifex

juanalfonsopr committed:
[FIX] l10n_es_aeat_mod303: Corregido cálculo de campo resultado_liquidacion y corregido formulario de mod303, sustituido campo casilla_46 repetido por campo casilla_69.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Añadir documentos explicativos del modelo

pedrobaeza committed:
[IMP] l10n_es_aeat_vat_prorrate: Enfoque definitivo

======================
AEAT - Prorrata de IVA
======================

Módulo para gestionar la prorrata del IVA en las declaraciones de la AEAT (por
el momento, en el modelo 303), según el artículo 92 de la Ley 37/1992, de 28 de
diciembre, del Impuesto sobre el Valor Añadido.

Uso
===

Realizando el modelo 303, se puede seleccionar si se desea aplicar prorrata de
IVA.

Prorrata general
----------------

En el caso de seleccionar "Prorrata general", aparece un nuevo campo llamado
"Porcentaje de prorrata de IVA" que puede ser rellenado a mano.

Además, cuando se realice el asiento de regularización, la parte proporcional
de la cuota no deducida se compensará como gasto.

Se ha incluido un calculador del porcentaje de prorrata, utilizable en 2 casos:

* Al comienzo de un nuevo ejercicio, para determinar el porcentaje provisional
  a aplicar hasta el último periodo de ese ejercicio.
* En la última declaración del ejercicio, para determinar el porcentaje
  definitivo de prorrata, y sobre el que se compensará la diferencia.

Problemas conocidos / Hoja de ruta
==================================

* La prorrata especial de IVA no está contemplada aún.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Algunos arreglos visuales

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat/l10n_es_aeat_mod303: Crear asiento de regularización cuando se hace la declaración

* Generador del asiento de regularización con cargo a un proveedor "Agencia
  Estatal de Administración Tributaria" creado al efecto.

Configuración
=============

Para activar la creación del asiento de regularización en un modelo, hay que
poner en el modelo correspondiente el campo allow_posting a True, y establecer
en la configuración de impuestos los conceptos que se regularizarán con el
flag "to_regularize".

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Don't ask for the bank account

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[ADD] l10n_es_aeat_mot303, casilla 77

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Mejor herencia de impuestos

El anterior método utilizado, al tener una clase y un many2one genérico,
no permitía poner los datos en múltiples modelos hijos, ya que se repetían
los IDs, y cogía los datos de un modelo en otros.

Con este nuevo sistema, se sigue utilizando una tabla, pero con doble
índice: el modelo y el ID, parecido a un campo reference, pero gestionado
por código propio en el método de cálculo.

El 303 está adaptado a este nuevo sistema, y se incluye un script de
migración que conserve los datos.

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Evitar problemas en campos related no readonly. Fixes #290

Cuando se tienen campos related, aunque el valor no se cambie,
Odoo lanza una escritura en el objeto relacionado cada vez que se
guarda el objeto principal. En el caso de la moneda asociada a la
compañía, hace que aquellos usuarios que no tengan permiso para
modificar la compañía (la mayoría), falle en la creación del modelo.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Export fields [27] and [45]

oca-transbot committed:
OCA Transbot updated translations from Transifex

omar7r committed:
[IMP] l10n_es_aeat_mod303: Se añade la gestión del iva diferido en el 303

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat_vat_prorrate: Include analytics + tests

oca-transbot committed:
OCA Transbot updated translations from Transifex

antespi committed:
[MIG] l10n_es_aeat/303: Rest of the migration

pedrobaeza committed:
[IMP] l10n_es_aeat: Several warnings and imps

pedrobaeza committed:
[MIG] l10n_es_aeat/mod303: Migration to 10.0

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Multi-company security rule

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt new BOE 2017 format

pedrorgil committed:
[FIX] aeat_mod_303. Corrección tipos de exportación (#465)

fuentes73 committed:
[FIX] l10n_es_aeat_mod303: Añadir la cuenta bancaria en la exportación

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Show compensate flag

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Record design link

pedrobaeza committed:
[FIX] l10n_es_aeat: Avoid multi-company error due to depends

Debido a que se utiliza una sola tabla para almacenar las líneas
de los impuestos de todos los modelos, hay una limitación en el ORM de Odoo
cuando se coloca el campo one2many de dichas líneas (tax_line_ids) como
dependencia en la definición del cálculo de un campo (entrada con
@api.depends), que recalcula los campos calculados de todos los modelos con el
mismo ID que el del registro en curso, lo que puede ser un problema en entornos
multi-compañía. Una solución a ello (aunque no evita el recálculo), es poner en
esos campos calculados `compute_sudo=True`.

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Conditional expression is reversed + 303 with conditional

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Change end year in previous export format

gurneyalex committed:
[REF] i18n: Updating translation terms from weblate [ci skip]

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10_es_aeat_mod303: Import 2017 data file

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Set date_end on 2014 export config

oihane committed:
[MIG] l10n_es_aeat_mod303: Migration to 11.0

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Automatic compensation and return improvement

Si no estás inscrito en el REDEME (Registro de Devolución Mensual), y la declaración
sale negativa, la única posibilidad es la de compensar, por lo que no tiene sentido
que se tenga que marcar manualmente la casilla, ya que si olvida y se exporta, Hacienda
no lo admite.

Por eso, este cambio detecta automáticamente esta situación y coloca el tipo de resultado,
eliminando la casilla extra.

Se incluye script de migración para restituir correctamente los datos de anteriores
declaraciones.

También se hace obligatoria la cuenta bancaria cuando la declaración sea a devolver, ya
que es también un dato obligatorio al presentarlo a Hacienda.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Migration script

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Migration not needed

Due to compute non stored field

JordiBForgeFlow committed:
- Cambia cuenta de contrapartida en función del resultado.
- En a casilla de importes a compensar debe constar, EN POSITIVO, las cuotas a compensar procedentes de periodos anteriores.
- Incluye mensaje para advertir que existen cuotas a compensar en periodos anteriores.
- Borra campo 'bank_account_id', que está ya duplicado en 'partner_bank_id' e incluye script de migración.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt to latest 2017 BOE format

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: BOE 2018 format

oca-transbot committed:
OCA Transbot updated translations from Transifex

luismontalba committed:
[IMP] l10n_es_aeat_mod303: Descuentos_cuota_IVA

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Improve tests for handling negative price units

carlosdauden committed:
[FIX] l10n_es_aeat_mod303: dict.iteritems() don't exist in python3. Change iteritems() to iter(items())

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod303: Fix refinements

* No need to change test structure with create
* Optimize code
* Fix tests in l10n_es_vat_prorrate
* Translations
* Other minor things

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Perform strict search of counterpart account

Odoo's operator "like" encloses the text to search into "%s", making
the search to match with accounts like "157000" and others as addition
to the expected "570000".

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Fix result when compensating

* Don't show the warning for draft reports
* When you have to return result, don't propagate the result to next report
* Put only once the warning

Abranes committed:
[FIX] l10n_es_aeat_mod303: Change export_type of Nombre to alphabetic

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

mtbochaca committed:
Translated using Weblate (català)

Currently translated at 100,0% (45 of 45 strings)

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_aeat_mod303/ca/

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Include negative amounts in refund tax amounts

They were ignored previously

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: Migration to 12.0

* Standard procedure
* Adapted tests
* README by fragments

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Add support for 2018 4T:

* BOE format
* Exonerated from 390

  * Fields
  * Mappings
  * BOE export

* README update
* Little change in AEAT engine
* Tests adapted + expanded

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Proper view when not exonerated

As numbers have changed, the view was not adjusted.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Don't inform 347 if not exonerated or not last period

Inform this field only in the case where this proceeds.

pedrobaeza committed:
[IMP] l10n_es_aeat: Add active field to export configs

This way, we don't initially see all the subconfigs for each model and we can concentrate
on main export configurations. This is got through `active` field default behaviour.
Archived config are still directly searchable through the subconfig selector in the
configuration line.

All models export configs have been adapted to this new approach.

pedrobaeza committed:
[IMP] l10n_es_aeat: Add chatter + activities to AEAT models

* Include chatter (mail.thread)
* Include activities (mail.activity.mixin)
* Track AEAT report state change
* Include new admin user in AEAT group by default

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 32.3% (54 of 167 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_mod303/es/

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Include both signs (debit and credit) in tax fee fields

Given this regular invoice:

1, IVA 10%, 10 €, 1 € tax
2, IVA 21%, -1 €, -0,21 € tax

that can happen when you return some packages (like returnable bottles), you have
a valid invoice with total positive amount to due, but in 303, the -0.21 € of the
tax fee is not included as there's no other positive tax fee to compensate that
negative amount, and the tax map line indicated that only credit operations are
included.

As the move type is already checked for avoiding to include refunds, we can change
to include both move lines types (debit and credit) for this. In fact, in the
base, both are already included.

As this can happen as well with other taxes and with supplier invoices, all
involved fields are changed in the same commit.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Add field 84

It's the sum of all the year of field 61, so the definition is the same.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt test to previous change

Field 88 doesn't include the new field 84

MiquelRForgeFlow committed:
[UPD] Remove deprecated from_string()

pedrobaeza committed:
[IMP+FIX] l10n_es_aeat: Move get taxes method to company + cleaning in tests

This way, we can reuse it in other modules that don't inherit AEAT report like SII.

We also clean a non needed reading of the tax description field.

Tests were still using description method for populating data, so we switch to new
system and perform needed changes in dependent tests.

s

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Avoid error importing BOE due to developer VAT

AEAT now forces to be a proper VAT format. We adjust at the same time the version
and some old year references.

acysos committed:
[FIX] l10n_es_aeat_mod303: fix 303 map for field number 60

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod303: Fix tests for fields 60/61 + improve ROADMAP

angelmoya committed:
[FIX] 303,facturas rectivicativas en 93 y 94

rlizana committed:
[12.0] l10n_es_aeat_*: [FIX] Caracteres no válidos 'NIF Empresa Desarrollo'

omar7r committed:
[FIX] l10n_es_aeat_mod303: Se añade el iva exento repercutido a la casilla 60. fix #1300

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 32.9% (55 of 167 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_mod303/es/

ValentinVinagre committed:
[IMP] l10n_es_aeat_mod303: black, isort, prettier

ValentinVinagre committed:
[MIG] l10n_es_aeat_mod303: Migration to 13.0

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_aeat_mod303/

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (158 of 158 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_aeat_mod303/es/

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (158 of 158 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_aeat_mod303/es/

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt odoo/odoo#58062

Now "IVA Exento No Sujeto" is not included in field 60, but it should be included
in field 83 for period 12 (December) in 303 for those exempted of the 390.

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

OCA-git-bot committed:
[UPD] README.rst

ozono committed:
[IMP] l10n_es_aeat_mod303: black, isort, prettier

ozono committed:
[MIG] l10n_es_aeat_mod303: Migration to 14.0

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: New 2021 BOE format

- Requires new field for not compensating full pending fee.
- New BOE format.
- Removed incorrect es_ES translation.
- Updated and completed Spanish translation.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Readonly for exonerated/volume on computed state

This way, we avoid the misunderstanding of changing that but not
computing fields 76 to 99.

Relates to #1553

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Fix fields 84 and 94

- Added missing tax "No sujeto (Bienes)" in field 84.
- Removed tax "Exportación (Bienes)" from field 94 (already on 84).

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Fix tests for the previous commit

Follow-up of 3a55cf3398e84ff5a49381a378b7bc11639ab717

Fix tests for the change.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Clean obsolete BOE export formats

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Change remaining taxes with only one sign

Remaining pieces of 6c3a6f2.

At the end, the refund part is controlled by move_type, not by balance
sign.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_aeat_mod303/

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Write fees to compensate together

If not, the constraint is evaluated each time, and can lead to
inconsistent states.

Fixes #1624

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Clarification about taxes accounts

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 98.7% (160 of 162 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_aeat_mod303/es/

pedrobaeza committed:
[FIX l10n_es_aeat_mod303: Proper search of counterpart account

The operator like searches for any coincidence, so it can select
account 147000 instead of 470000. We switch to =like operator for
avoiding the problem.

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_aeat_mod303/

angelmoya committed:
[IMP] l10n_es_aeat_mod303: Add support for new BOE 2021/07 format
    * New tax mapping, field 84 now split on 84 and 125
    * New export config, now in csv
    * New marca_sepa field
    * Update field 80

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod303: Finish 2021-07 format + v14 adaptation

* Transform tax map to CSV
* Incorrect BOE export definition
* Simplify certain expressions
* Move CSV to subdirectory to classify them by format change batch
* End date for previous BOE format
* Fix marca_sepa computation
* Restore back field 85 for older reports

acysos committed:
[FIX] l10n_es_aeat_mod303: Incorrect taxes + marca_sepa

Remove taxes from field 61, and add them to both fields 120 y 122.
That's because since 2021/07, these fields contains such values, and as
the tax mapping is only used since this date, it's safe to do it.

Something similar applies to marca_sepa, as even if doing a previous
report, this field is not used at all, so we can remove the initial
condition and put the value only when performing return.

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Proper depends for compute_sepa

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: 2021/07 format: Don't populate values if no return

This hasn't been detected before, as testing reports didn't have bank
account populated, but on a new one, it seems AEAT doesn't allow to have
any content in these fields if the result is not to return.

oca-travis committed:
[UPD] Update l10n_es_aeat_mod303.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_aeat_mod303/

ValentinVinagre committed:
[IMP] l10n_es_aeat_mod303: black, isort, prettier

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Don't populate field 61

This field is deprecated since 07/2021, and bringing it, even empty, can
be confusing for users, so better to remove it.

ValentinVinagre committed:
[MIG] l10n_es_aeat_mod303: Migration to 15.0

Commit:
[UPD] Update l10n_es_aeat_mod303.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Last BOE export inaccuracies

- Don't apply sign to "Volumen operaciones"
- Don't export SWIFT if Spanish or European bank

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: New BOE format 2022

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Illegal quoting in export line

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Marca SEPA flag should be w/o sign

TT34690

pedrobaeza committed:
[FIX] black formatting

juliap98 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (168 of 168 strings)

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_aeat_mod303/ca/

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Avoid config name clash

MiquelRForgeFlow committed:
[FIX] l10n_es_aeat_*: exceptions.Warning -> exceptions.UserError

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Include new tax 5%

Following https://github.com/odoo/odoo/pull/95967, the mapping is done
for mapping the corresponding fields in the 303 model.

If someone sells electricity at that rate, there's no mapping yet, as we
don't know how the AEAT is going to reformat the model.

TT38004

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Avoid rounding problems

When doing mathematical operations over several rounded amounts, you can
get a non rounded result. Example:

>>> tax_lines.mapped("amount")
[1022.18, 125899.42, 5399.41, 7354.03, 30.5, -3104.6, -181.45, 0.27, 18.03, 0.0, -0.34]
>>> sum(tax_lines.mapped("amount"))
136437.44999999995

Leading to possible wrong report result ("Solicitud de devolución"
instead of "Resultado cero").

In this commit, we round all the amounts + use float_compare for the
result to avoid the problem.

TT40670

FernandoRomera committed:
[16.0][MIG] l10n_es_aeat_mod303: Migration to 16.0

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Convert 303 activity codes data to CSV

For faster load and maintenance

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt to 2022 exonerated reqs

- Activity codes are now 3 letters long instead of integers.
- New activity codes for this year, and shared for 4T and 12.
- Implemented new date_start and date_end on activity codes, period_type
  not required and code as char, along with an extended domain on 303
  report for accomodating to the new specs.
- Fix BOE format to real specs.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Domain for other activity codes

They were not adjusted to new domain and thus, you can't select the
proper activity codes for this year.

TT41421

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: New BOE format for 2023

TT41657

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Don't put bank details if not foreign

AEAT is now stricter forcing to not fill this information if the bank
account is not a foreign one.

TT41873

Commit:
[FIX] l10n_es_aeat_mod303: Add new result_types to domain

omar7r committed:
[IMP] l10n_es_aeat_mod303: Include ISP investments taxes

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 3.8% (7 of 180 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_mod303/sl/

Commit:
Translated using Weblate (Catalan)

Currently translated at 88.8% (159 of 179 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_mod303/ca/

JordiBForgeFlow committed:
[FIX] l10n_es_aeat_mod303: update map tax line for field number 31.

In case of investment operations with reverse charge the field 31 was not reflecting
any amount.

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_mod303/es/

Commit:
[FIX] l10n_es_aeat_mod303: se elimina el mapeo del impuesto 'No sujeto Repercutido (Servicios)' y 'No sujeto Repercutido (Bienes)' de la linea 60 y se agregan a la linea 120

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_mod303/es/

EmilioPascual committed:
[IMP] l10n_es_mod_303: Cuota a compensar (casilla 110)

Calculate potential_cuota_compensar with sum 110 and 71 of the previous report

MT-4519

Co-authored-by: Arantxa Sudón <104626993+ArantxaSudon@users.noreply.github.com>
Co-authored-by: Loida Vilaplana <104626797+loida-vm@users.noreply.github.com>
Co-authored-by: Rafael Blasco <8736623+rafaelbn@users.noreply.github.com>

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: pre-commit auto fixes

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: Migration to 17.0

- Standard procedure.
- Converted tax map lines to new format.
- Converted views to new readonly/required/invisible syntax.
- Adapted tests.

TT48475

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-17.0/l10n-spain-17.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-17-0/l10n-spain-17-0-l10n_es_aeat_mod303/

yajo committed:
[FIX] l10n_es_aeat_mod303: comunicar IBAN en ingresos

Cuando el 303 sale a ingresar, si el usuario indica una cuenta bancaria SEPA, se puede comunicar para domiciliarlo automáticamente.

La letra correspondiente al tipo de resultado sería la "U" (domiciliación del ingreso en CCC), pero resulta que esa letra nunca se usaba.

Ahora el resultado tipo "I" se convierte en "U" si el usuario indica cuenta bancaria.

Fix https://github.com/OCA/l10n-spain/issues/3514
@moduon MT-5662

Co-authored-by: <104626797+loida-vm@users.noreply.github.com>

yajo committed:
[FIX] l10n_es_aeat_mod303: soporte para cuenta tributaria

Al modelo le faltaba el concepto de [cuenta corriente tributaria][1], así que lo he añadido. Esta cuenta corriente, a pesar del nombre, no es una cuenta bancaria; es un instrumento fiscal para gestionar la deuda de forma más ágil.

Por lo tanto, cuando el usuario escoge esa opción, la opción de cuenta bancaria desaparece y la marca SEPA se vuelve 0. También cambia el tipo a "V" cuando es a devolver.

[1]: https://sede.agenciatributaria.gob.es/Sede/procedimientoini/GZ31.shtml?faqId=9767d0c33063c710VgnVCM100000dc381e0aRCRD

@moduon MT-5662

EmilioPascual committed:
[IMP] l10n_es_aeat_mod303: Return last period

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (180 of 180 strings)

Translation: l10n-spain-17.0/l10n-spain-17.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-17-0/l10n-spain-17-0-l10n_es_aeat_mod303/es/

Tisho99 committed:
[IMP]l10n_es_aeat_mod303: add dua taxes

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: New declaration format for 2024-10

- New fields 165-170.
- Mapping all the new taxes.
- New BOE format.

TT51084

etobella committed:
[IMP] l10n_es_aeat_mod303: Mark the box only on complementary declarations

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 99.4% (181 of 182 strings)

Translation: l10n-spain-17.0/l10n-spain-17.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-17-0/l10n-spain-17-0-l10n_es_aeat_mod303/es/

ljsalvatierra-factorlibre committed:
[FIX] l10n_es_aeat_mod303: casilla 17 modelo 2024-10

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Missing "to regularize" flags

IVA 2% + Recargo equivalencia 0.26% should be included in VAT
regularization entry.

TT52149

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Fields 14, 15 and 80 miss the new taxes

EmilioPascual committed:
[FIX] l10n_es_aeat_mod303: Error generación asiento contable cuando sale a pagar pero se compensa parcialmente con ejercicios anteriores

@moduon MT-8455

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Follow 2 column layout in exonerated 390

If not, the fields are stretched on regular screens, and they can't be
filled.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Remove field 96

It was incorrectly mapped to "recargo de equivalencia" base amounts,
which is not correct, as that base amounts are already included in
regular operations, so it appears duplicated.

This field is for being filled in specific cases where the subject is
on the special fiscal regime "Recargo de equivalencia".

The tests have been changed, as now field 88 doesn't contain this field.

TT52996

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Missing taxes in 25-26

Recargo equivalencia 0,26% and 1% for refunds are missing.

TT54712

pedrobaeza committed:
[IMP] l10n_es_aeat*: Show only fields with values by default

Models like 303 and 390 have a lot of fields, so showing them all even
if the amount is 0 can be counterproductive. Thanks to the magic of
one2many fields, we can continue calculating all the fields and showing
them for traceability purposes, but filter out in another o2m field
those with amount with hardly no cost, and now 2 tabs are shown for
these lines, having precedence the filtered one.

victoralmau committed:
[FIX] l10n_es_aeat_mod303: Fix tests related to S_IVA21ISP

Tax S_IVA21ISP was removed at https://github.com/odoo/odoo/commit/401a961258e991151370899d4b61809879d5a5a8 therefore, the test data is incorrect.

If tax S_IVA21ISP is restored, this change must be reverted.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: New BOE format 2025

In fact, the changes are very small for the implemented things, as
the changes on the REAGP, which are the biggest, are not implemented.
We just need to force to 0 the percentages for the ephemeral taxes.
They just can ignore that fields for not bothering people...

There's also a move for "Recargo equivalencia 0.5%" from 16-18 to
168-170 inadvertently.

AaronHForgeFlow committed:
[MIG] l10n_es_aeat_mod303: Migration to v18

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Avoid negatives in fee to compensate

If the previous 303 report has remaining fee to compensate, and the
result of the report is positive and bigger than the fee to compensate,
when computing the current one, the constraint

  "The fee to compensate must be indicated as a positive number."

will raise always.

To avoid it, two changes have been done:

1. Only substract report result if it's of type C.
2. Nullify any possible result when it's less than 0.

Amendment of 4bd1c0280cbdaf7f452435d398fe5000ec344a6f

TT55245

pedrobaeza committed:
[FIX] l10n_es_*: Fix intra/extra-community taxes refs and new exempt taxes

Odoo screwed up the tax XML-IDs in the middle of the 17.0 lifecycle,
introducing an unfolding of the exempt taxes, using existing ones
that belongs to the sales intra and extra community taxes.

After that, they have renamed the XML-IDs of both taxes to new ones.

We need to handle this in 2 ways:

- Detect and correct existing installations previous to this
  taxpocalypse for renaming them.
- Change the reference in all places to the new XML-IDs.

Besides, we have to add the unfolded exempt taxes in the reports.

acysos committed:
[IMP] l10n_es_aeat*: cambio de menu a declaraciones con submenus por agencia

[IMP] l10n_es_aeat*: cambio de menu a declaraciones con submenus por agencia

[IMP] l10n_es_aeat*: cambio de menu a declaraciones con submenus por agencia

Commit:
[UPD] Update l10n_es_aeat_mod303.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update l10n_es_aeat_mod303.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-18.0/l10n-spain-18.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-18-0/l10n-spain-18-0-l10n_es_aeat_mod303/

Tisho99 committed:
[IMP] l10n_es_aeat_mod303: add dua taxes to 2024-10 format

OCA-git-bot committed:
[BOT] post-merge updates

EmilioPascual committed:
[FIX] l10n_es_aeat_mod303: Use case [87] Reamining fees to compensate is equal to 0 in previous report.

In the previous report when box 87  is equal to 0, result is 'To compensate' and box 71 is negative, then box 110 must also contain pending fees to compensate.

@moduon MT-10979

MRomeera committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (183 of 183 strings)

Translation: l10n-spain-18.0/l10n-spain-18.0-l10n_es_aeat_mod303
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-18-0/l10n-spain-18-0-l10n_es_aeat_mod303/es/

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Remove missing tax from tests

Odoo has removed some taxes in
https://github.com/odoo/odoo/commit/d3c7e51e94d98da9086a3817b157c4e125c80790,
so although we keep the XML-ID definition in map taxes (for backwards
compatibility), tests should be adapted for not using that tax.

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

Reyes4711-S73 reviewed (approved):
LGTM
Functional Review OK

pablo-cort-s73 reviewed (approved):
LGTM

dalonsod committed:
[FIX] l10n_es_aeat_mod303: tener en cuenta redondeo en comprobaciones cuotas compensación

En casos límite, como puede ser establecer la compensación pendiente como
exactamente la cuota a compensar aplicada, podía impedir la operación por
problemas de redondeo.

acysos commented:
/ocabot migration l10n_es_aeat_mod303

dalonsod commented:
@sbiosca-s73 ¿puedes añadir el siguiente fix?

*  #4748

sbiosca-s73 commented:
@dalonsod hecho!

Andrii9090-tecnativa committed:
[IMP] l10n_es_aeat_mod303: New declaration format for 2026
- New fields 108,109,112.

TT60927

Andrii9090-tecnativa committed:
[FIX] l10n_es_aeat_mod303_vat_prorate: Adjust it to the specifications 2026

TT60927

rafaelbn commented:
Hola @sbiosca-s73 ,

¿Puedes por favor hacer un rebase a ver si conseguimos que runbot fucnione y lo probamos? Ahora mismo no funciona.

<img width="1701" height="879" alt="imagen" src="https://github.com/user-attachments/assets/741a0170-4e3e-4a47-bc3a-0c1d055f7ded" />

Muchas gracias

MT-14012 @moduon

sbiosca-s73 commented:
@rafaelbn hecho, parece que ya está accesible. Muchas gracias!

ArantxaSudon reviewed (approved):
Revisión funcional
He probado con impuestos que vayan a las diferentes casillas y todas funcionan correctamente

<img width="836" height="910" alt="Image" src="https://github.com/user-attachments/assets/4a52ed52-0007-4e8f-9720-a90a7bf84b9f" />

Gracias @sbiosca-s73 

MT-14012 @moduon

EmilioPascual commented:
¿@sbiosca-s73 puedes añadir los últimos commits? Así se puede probar todos los cambios referentes a 2026.
Además, ya puedes quitar la dependencia con el PR de migración de `l10n_es_aeat` puesto que ya se ha fusionado.
Gracias.

sbiosca-s73 committed:
[IMP] l10n_es_aeat_mod303: pre-commit auto fixes

sbiosca-s73 committed:
[MIG] l10n_es_aeat_mod303: Migration to 19.0

Remove tax: s_iva21isp, s_iva0_e, s_iva0_ic

sbiosca-s73 commented:
@EmilioPascual cambios hechos! Muchas gracias.

EmilioPascual commented on l10n_es_aeat_mod303/data/2024-10/l10n.es.aeat.map.tax.line.csv:
```diff
@@ -0,0 +1,62 @@
+aeat_mod303_2024_10_map_line_03,aeat_mod303_2024_10_map,3,Régimen General - Cuota 4%,1,regular,amount,both,0,,"s_iva4b,s_iva4s"
+aeat_mod303_2024_10_map_line_04,aeat_mod303_2024_10_map,4,Régimen General - Base imponible 10%,0,regular,base,both,0,,"s_iva10b,s_iva10s"
+aeat_mod303_2024_10_map_line_06,aeat_mod303_2024_10_map,6,Régimen General - Cuota 10%,1,regular,amount,both,0,,"s_iva10b,s_iva10s"
+aeat_mod303_2024_10_map_line_07,aeat_mod303_2024_10_map,7,Régimen General - Base imponible 21%,0,regular,base,both,0,,"s_iva21b,s_iva21s,s_iva21isp"
```
Este impuesto `s_iva21isp` ya no existe desde hace algunas versiones ¿Habría que dejarlo por si alguien lo sigue usando? ¿O mejor quitarlo y si alguien lo usa que lo mapee a mano?

pedrobaeza replied:
Sí, vamos a aprovecha para quitarlo. Igualmente, habría que aprovechar para quitar formatos de exportación/cálculo de impuestos más antiguo de 1 año.

EmilioPascual replied:
Pues si aprovechamos a quitar, he detectado los siguientes que ya no están en 19:

- s_iva21isp
- s_iva0_e
- s_iva0_ic

En cuanto a quitar formatos antiguos, yo dejaría 4 años siempre. Al final es el plazo que se tiene para poder presentar posibles rectificativas.

sbiosca-s73 replied:
@EmilioPascual hecho! Gracias.

EmilioPascual reviewed (approved):
Buen trabajo con la migración @sbiosca-s73 
Revisado código y funcionalmente

pedrobaeza reviewed (approved):
/ocabot merge nobump
